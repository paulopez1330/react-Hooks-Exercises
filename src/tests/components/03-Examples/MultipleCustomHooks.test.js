import React, {  } from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-Examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Prueba de <MultipleCustomHooks />', () => {
    
    test('debe mostrarse correctamente', () => {

        useCounter.mockReturnValue({
            counter: 10,
            increment:() => {}
        });

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks /> );
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar informacion', () => {

        useCounter.mockReturnValue({
            counter: 1,
            increment:() => {}
        });
                
        useFetch.mockReturnValue({
            data: [{
                author: 'borges',
                quote: 'hola mundo'
            }],
            loading: false,
            error: null
        });        
        
        const wrapper = shallow( <MultipleCustomHooks /> );
        
        expect( wrapper.find('.alert').exists() ).toBe( false );
        expect( wrapper.find('.mb-0').text().trim() ).toBe( 'hola mundo' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'borges' );
    })    
})