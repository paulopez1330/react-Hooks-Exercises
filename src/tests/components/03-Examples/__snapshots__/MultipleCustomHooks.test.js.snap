// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <MultipleCustomHooks /> debe mostrarse correctamente 1`] = `
<div
  className="container-fluid"
>
  <h1>
    Breakingbad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente Quote
  </button>
</div>
`;
